integr8ly/integreatly-operator
# INTLY-2325 AMQ Online installation

There are two commits:
- 0e7689e for vendor and Gopkg lockfile updates
- e002e24 for code changes

EnMasse doesn't include definitions for a number of resources in its code, so they have been generated in our project. This seems to be due to `*Plans` and a few other resources being handled by Java code.

We also create and reconcile all of the default resources that are provisioned on the current version of Integreatly.

Verification:
- Create the following `OperatorSource` to point at my registry:
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  finalizers:
    - finalizer.operatorsources.operators.coreos.com
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: aidenkeating
  type: appregistry
```
- Create a `Subscription` for the Integreatly Operator
- Create an `Installation` in a `Namespace` the operator is watching
```
apiVersion: integreatly.org/v1alpha1
kind: Installation
metadata:
  name: example-installation
spec:
  type: workshop
  namespacePrefix: integreatly-
```
- Verify the `amq-online` `Namespace` is created as expected
  - You should be able to access the `Console` `Route` URL
  - You should be able to see `AddressPlans`, `AddressSpacePlans`, `AuthenicationServices`, `BrokeredInfraConfigs` and `StandardInfraConfigs` in the cluster
